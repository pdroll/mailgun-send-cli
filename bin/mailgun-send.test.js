const path = require('path');
const { exec } = require('child_process');

const cli = (args) => new Promise((resolve) => {
  exec(`node ${path.resolve('bin/mailgun-send.js')} ${args.join(' ')}`,
    { cwd: '.' },
    (error, stdout, stderr) => {
      resolve({
        code: error && error.code ? error.code : 0,
        error,
        stdout,
        stderr,
      });
    });
});

it('prints help text', async () => {
  const result = await cli(['-h']);

  expect(result.code).toBe(0);
  expect(result.stdout).toMatchSnapshot();
});
