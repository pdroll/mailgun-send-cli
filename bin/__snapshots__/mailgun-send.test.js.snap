// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints help text 1`] = `
"Usage: mailgun-send [options]

You will be prompted to enter your Mailgun API Key [https://app.mailgun.com/app/account/security/api_keys] and Domain [https://app.mailgun.com/app/sending/domains/] on your first use.

Options:
  -V, --version           output the version number
  -s, --subject <value>   Subject of Email
  -t, --to <value>        Email address of recipient of email
  -f, --from <value>      Email address of email sender
  -r, --reply <value>     ReplyTo email address. Optional
  -c, --cc <value>        Email address to CC. Optional
  -b, --bcc <value>       Email address to BCC. Optional
  -T, --text <value>      Text to send as body of email. Must specify this or
                          --htmlpath.
  -H, --htmlpath <value>  Path to HTML file to send as email. Must specify this
                          or --text.
  -R, --reset             Reset Mailgun API key and Domain. You will be
                          prompted to enter these again.
  -v, --verbose           Output more detailed information, such as message id
  -h, --help              display help for command
"
`;
